src/engine/ResultTable.cpp

```diff
@@ -1,17 +1,61 @@
-// Copyright 2015, University of Freiburg,
-// Chair of Algorithms and Data Structures.
-// Author: Björn Buchhold (buchhold@informatik.uni-freiburg.de)
+// Copyright 2015 -2022, University of Freiburg
+// Chair of Algorithms and Data Structures
+// Authors: Björn Buchhold <b.buchhold@gmail.com>
+//          Hannah Bast <bast@cs.uni-freiburg.de>
 
-#include "./ResultTable.h"
+#include "engine/ResultTable.h"
 
 #include <cassert>
 
+#include "global/Id.h"
+#include "global/ValueId.h"
+
+// _____________________________________________________________________________
+Id LocalVocab::getIdAndAddIfNotContained(const std::string& word) {
+  if (constructionHasFinished_) {
+    throw std::runtime_error(
+        "Invalid use of `LocalVocab`: You must not call "
+        "`getIdAndAddIfNotContained` after `endConstructionPhase` has been "
+        "called");
+  }
+  // The following code avoids computing the hash for `word` twice in case we
+  // see it for the first time (note that hashing a string is not cheap). The
+  // return value of the `insert` operation is a pair, where `result.first` is
+  // an iterator to the (already existing or newly inserted) key-value pair, and
+  // `result.second` is a `bool`, which is `true` if and only if the value was
+  // newly inserted.
+  auto result = wordsToIdsMap_.insert(
```

Comment:
`auto [iterator, isNewWord] = wordsToIdsMap_.insert(...` makes the following lines much more readable:)

Reply:
Done